### PR TITLE
refactor: refactor functions over 100 lines and complexity over 20

### PR DESCRIPTION
**Pull request by AlisonB319**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by AlisonB319** [comment]
@alexeldeib @cameronmeissner Here is PR2

File: pkg/agent/utils.go [hunk 1]
```diff
@@ -362,12 +362,7 @@ func GetTLSBootstrapTokenForKubeConfig(tlsBootstrapToken *string) string {
 	return *tlsBootstrapToken
 }
 
-// GetKubeletConfigFileContent converts kubelet flags we set to a file, and return the json content.
-func GetKubeletConfigFileContent(kc map[string]string, customKc *datamodel.CustomKubeletConfig) string {
-	if kc == nil {
```
Comment (alexeldeib) [inline pkg/agent/utils.go:367]: hmm, this serializes differently?

Comment (alexeldeib) [reply pkg/agent/utils.go:367]: jk the func moved

File: pkg/agent/utils.go [hunk 2]
```diff
@@ -362,12 +362,7 @@ func GetTLSBootstrapTokenForKubeConfig(tlsBootstrapToken *string) string {
...
-		return ""
-	}
-	// translate simple values.
+func GetAKSKubeletConfiguration(kc map[string]string) *datamodel.AKSKubeletConfiguration {
```
Comment (alexeldeib) [inline pkg/agent/utils.go:365]: private?
```suggestion
func getAKSKubeletConfiguration(kc map[string]string) *datamodel.AKSKubeletConfiguration {
```

File: pkg/agent/utils.go [hunk 3]
```diff
@@ -397,41 +392,12 @@ func GetKubeletConfigFileContent(kc map[string]string, customKc *datamodel.Custo
...
-	kubeletConfig.FeatureGates = strKeyValToMapBool(kc["--feature-gates"], ",", "=")
-
-	// system reserve and kube reserve.
-	// looks like "cpu=100m,memory=1638Mi".
-	kubeletConfig.SystemReserved = strKeyValToMap(kc["--system-reserved"], ",", "=")
-	kubeletConfig.KubeReserved = strKeyValToMap(kc["--kube-reserved"], ",", "=")
-
-	// Settings from customKubeletConfig, only take if it's set.
+//nolint:gocognit
+func SetCustomKubeletConfig(customKc *datamodel.CustomKubeletConfig,
```
Comment (alexeldeib) [inline pkg/agent/utils.go:399]: private?
```suggestion
func setCustomKubeletConfig(customKc *datamodel.CustomKubeletConfig,
```

**Review by alexeldeib — APPROVED** [review]
think this lgtm, cc @cameronmeissner for kubelet config stuffs

**Comment by alexeldeib** [comment]
stray mariner e2e failure?

File: pkg/agent/params.go [hunk 4]
```diff
@@ -102,6 +103,70 @@ func assignKubernetesParametersFromAgentProfile(profile *datamodel.AgentPoolProf
 	}
 }
 
+func assignKubernetesParametersfromKubernetesConfig(kubernetesConfig *datamodel.KubernetesConfig, properties *datamodel.Properties,
```
Comment (cameronmeissner) [inline pkg/agent/params.go:106]: looks like some of this stuff is being touched by both PRs

**Pull request merged**